react/utils/CurrentPosition.js

```diff
@@ -1,37 +1,38 @@
 import { geolocationAutoCompleteAddress } from 'vtex.address-form/geolocationAutoCompleteAddress'
 
 export function getCurrentPosition(successCallback, errorCallback) {
-  return navigator.geolocation.getCurrentPosition(
-    position => successCallback(position),
-    error => {
-      // TODO#1: retry in case of timeout, possibly with a lower accuracy.
-      // Not implementing right now because we might need to look into issues that long
-      // timeouts might cause, especially in terms of UX. But below is a suggestion:
-      /*
-      const TIMEOUT = 3
-      if (error.code === TIMEOUT ) {
-        navigator.geolocation.getCurrentPosition(successCallback, errorCallback, {
-          maximumAge: Infinity,
-          timeout: 20000,
-          enableHighAccuracy: false,
-        })
-      } else {
-        errorCallback(error)
-      }
-      */
-      errorCallback(error)
-    },
+  let hasBeenCanceled = false
+
+  const handleSuccess = position => {
+    if (hasBeenCanceled) {
+      return
+    }
+    successCallback(position)
+  }
+
+  const handleError = error => {
+    if (hasBeenCanceled) {
+      return
+    }
+    errorCallback(error)
+  }
+
+  navigator.geolocation.getCurrentPosition(
+    handleSuccess,
+    handleError,
     {
       maximumAge: Infinity,
-      // Increased timeout from 2000 to 6000.
-      // 2000 was timing out on my (@lbebber) machine, but increasing to ~5000 did the trick.
-      // Setting to 6000 for a balance between leaving a margin vs not waiting too much, but
-      // the ideal timeout value has to be looked into.
-      // Refer to TODO#3
-      timeout: 6000,
+      // getCurrentPosition timeout varies a lot depending on device,
+      // browser, and even user location. Avoid reducing this value,
+      // but if you do, track timeouts on a logging service.
+      // (currently, look for "dismissedGeolocation" on our kibana logs)
+      timeout: 10000,
```

Comment:
Increased the timeout here, this bit is important

Reply:
I don't know how the project are handled with constants, but I think that is good to put this value in a variable.

Reply:
in this case I think I prefer to keep the value inline, because it's never used elsewhere, it's arbitrary, and it's very specific to this function.

keeping it close to the function helps giving it context I think, and makes it easier to track it down and update if necessary